### Measures of Dispersion
Measures of dispersion tell us how much the data values are spread out or scattered around the center.

For example:

Dataset A: 48, 49, 50, 51, 52
Dataset B: 20, 35, 50, 65, 80

Both have the same mean = 50, but Dataset B has much more variation.

### Why is it Important?
* Understand data variability.
* Compare consistency between datasets.
* Detect outliers.
* Help in feature scaling and model evaluation.
* Improve decision-making in business and AI/ML.

### Range
Definition

Range is the difference between the maximum and minimum values.

Formula
Range=Maximum−Minimum

In [4]:
marks = [45, 50, 55, 60, 65, 70, 75, 80, 85, 90]

print(max(marks) - min(marks))

45


### Example:
A teacher wants to know the difference between the highest and lowest marks in a class.

### Variance
Variance measures how far the data values are spread from the mean.

Formula:

Variance = Σ(x − Mean)² / N

Mean = 67.5

Variance ≈ 206.25

Where:

* \(x_i\) = individual value
* \(\mu\) = mean
* \(N\) = number of observations

In [12]:
import numpy as np

data = [2, 4, 6]

variance = np.var(data)

print("Variance:", variance)

Variance: 2.6666666666666665


### Standard Deviation

Standard deviation is the square root of variance.

It tells us the typical amount by which values differ from the mean.

Formula
$$ \sigma = \sqrt{\frac{\sum(x_i-\mu)^2}{N}} $$

In [15]:
marks = [45, 50, 55, 60, 65, 70, 75, 80, 85, 90]

print(np.std(marks))

14.361406616345072


In [19]:
#AIML
from sklearn.preprocessing import StandardScaler

X = [[10], [20], [30], [40], [50]]

scaler = StandardScaler()

X_scaled = scaler.fit_transform(X)

print(X_scaled)

[[-1.41421356]
 [-0.70710678]
 [ 0.        ]
 [ 0.70710678]
 [ 1.41421356]]


Mean = 0,
Standard deviation = 1

### Quartiles
Definition

Quartiles divide ordered data into four approximately equal parts.

There are three important quartiles:
| Quartile | Meaning                  |
| -------- | ------------------------ |
| Q1       | 25th percentile          |
| Q2       | 50th percentile / Median |
| Q3       | 75th percentile          |


In [24]:
import numpy as np

data = [10, 20, 30, 40, 50, 60, 70, 80]

Q1 = np.percentile(data, 25)
Q2 = np.percentile(data, 50)
Q3 = np.percentile(data, 75)

print("Q1:", Q1)
print("Q2:", Q2)
print("Q3:", Q3)

Q1: 27.5
Q2: 45.0
Q3: 62.5


### AI/ML Example
Quartiles are useful for:

* Detecting outliers
* Understanding feature distributions
* Exploratory Data Analysis
* Box plots

### Percentiles
A percentile tells us the value below which a certain percentage of observations falls.

For example:

90th percentile = 90% of observations are at or below that value.

Here,

75th Percentile = 78.75

This means 75% of students scored at or below 78.75 marks.

In [31]:
import numpy as np

data = [10, 20, 30, 40, 50, 60, 70, 80, 90, 100]

p90 = np.percentile(data, 90)

print("90th Percentile:", p90)

90th Percentile: 91.0


In [36]:
threshold = np.percentile(data, 95)

print("95th percentile:", threshold)

95th percentile: 95.49999999999999


AI/ML Example

Percentiles can be used for:

* Outlier detection
* Feature analysis
* Data preprocessing
* Model prediction analysis
* Latency monitoring

### Coefficient of Variation
Coefficient of Variation (CV) compares the standard deviation with the mean.

Coefficient of Variation measures the relative amount of variation compared with the mean.

It is especially useful when comparing datasets with different units or different means.
It is expressed as a percentage.

Formula:

CV = (Standard Deviation / Mean) × 100

Here,

Mean = 67.5

Standard Deviation = 14.36

CV = (14.36 / 67.5) × 100

CV ≈ 21.27%

In [41]:
import numpy as np

data = [10, 20, 30, 40, 50]

mean = np.mean(data)
std = np.std(data)

cv = (std / mean) * 100

print("Mean:", mean)
print("Standard Deviation:", std)
print("CV:", cv, "%")

Mean: 30.0
Standard Deviation: 14.142135623730951
CV: 47.14045207910317 %


In [43]:
import numpy as np
import pandas as pd

data = [10, 12, 15, 18, 20, 22, 25, 30, 100]

# Central tendency
mean = np.mean(data)
median = np.median(data)

# Dispersion
data_range = np.max(data) - np.min(data)
variance = np.var(data)
std = np.std(data)

# Quartiles
Q1 = np.percentile(data, 25)
Q3 = np.percentile(data, 75)

# IQR
IQR = Q3 - Q1

# CV
cv = (std / mean) * 100

print("Mean:", mean)
print("Median:", median)
print("Range:", data_range)
print("Variance:", variance)
print("Standard Deviation:", std)
print("Q1:", Q1)
print("Q3:", Q3)
print("IQR:", IQR)
print("CV:", cv)

Mean: 28.0
Median: 20.0
Range: 90
Variance: 682.8888888888889
Standard Deviation: 26.132142830026183
Q1: 15.0
Q3: 25.0
IQR: 10.0
CV: 93.3290815358078
